### PR TITLE
[fix] - MV AM Unit test to golden

### PR DESCRIPTION
**Pull request by heliapb**
## Description
From issue: https://github.com/prometheus-operator/prometheus-operator/issues/7193
Move all AM to use golden. 
All golden files were generated with the `go test -v ./pkg/alertmanager -run <test-case> -update`

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by heliapb** [comment]
@slashpai I think I got to all the tests, but let me know if anything needs adjustment.

File: pkg/alertmanager/amcfg_test.go [hunk 1]
```diff
@@ -78,15 +79,16 @@ func TestInitializeFromAlertmanagerConfig(t *testing.T) {
 
 	tests := []struct {
 		name            string
+		golden          string
```
Comment (slashpai) [inline pkg/alertmanager/amcfg_test.go:82]: nit: may be move golden as last field, we follow a similar way in https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg_test.go

File: pkg/alertmanager/amcfg_test.go [hunk 2]
```diff
@@ -78,15 +79,16 @@ func TestInitializeFromAlertmanagerConfig(t *testing.T) {
...
 		globalConfig    *monitoringv1.AlertmanagerGlobalConfig
 		matcherStrategy monitoringv1.AlertmanagerConfigMatcherStrategy
 		amConfig        *monitoringv1alpha1.AlertmanagerConfig
-		want            *alertmanagerConfig
 		wantErr         bool
 	}{
 		{
-			name: "valid global config",
+			name:   "valid global config",
+			golden: "valid_global_config.golden",
```
Comment (slashpai) [inline pkg/alertmanager/amcfg_test.go:91]: nit: lets set this as last field

**Review by slashpai — CHANGES_REQUESTED** [review]
overall looks good to me, just some nits

File: pkg/alertmanager/amcfg_test.go (hunk 1 above)
Comment (heliapb) [reply pkg/alertmanager/amcfg_test.go]: Thanks, I think I got to all the golden, pls review.

File: pkg/alertmanager/amcfg_test.go [hunk 3]
```diff
@@ -1004,8 +803,16 @@ func TestInitializeFromAlertmanagerConfig(t *testing.T) {
 				t.Errorf("initializeFromAlertmanagerConfig() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (slashpai) [inline pkg/alertmanager/amcfg_test.go:803]: nit: looks like we don't need this now as we are using require.Error in next step

File: pkg/alertmanager/amcfg_test.go [hunk 4]
```diff
@@ -1004,8 +803,16 @@ func TestInitializeFromAlertmanagerConfig(t *testing.T) {
...
 				return
```
Comment (slashpai) [inline pkg/alertmanager/amcfg_test.go:804]: this too

File: pkg/alertmanager/amcfg_test.go [hunk 5]
```diff
@@ -3619,8 +3164,11 @@ func TestHTTPClientConfig(t *testing.T) {
 				require.Error(t, err)
 				return
 			}
+
+			amConfigs, err := yaml.Marshal(tc.in)
 			require.NoError(t, err)
-			require.Equal(t, tc.expect, *tc.in)
+
```
Comment (slashpai) [inline pkg/alertmanager/amcfg_test.go:3170]: nit: extra blank line

**Review by slashpai — COMMENTED** [review]
overall good and thank you for your patience with this :)

I would like another set of eyes as there are many changes

@simonpasquier @nicolastakashi  can you take a look too?

**Review by simonpasquier — APPROVED** [review]
:partying_face: thanks a lot!

**Pull request merged**